module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -2258,4 +2246,42 @@ protected function collectRequestAccountStats(array $records): ?array
         }
         return $accountStatus;
     }
+
+    /**
+     * Add a message about any pending email change to the flash messenger
+     *
+     * @param \VuFind\Db\Row\User $user User
+     *
+     * @return void
+     */
+    protected function addPendingEmailChangeMessage($user)
+    {
+        if (!empty($user->pending_email)) {
+            // Check if the message has already been added
+            foreach ($this->flashMessenger()->getInfoMessages() as $message) {
```

Comment:
Is this necessary? The view helper should already be deduplicating flash messages [here](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/View/Helper/Root/Flashmessages.php#L86). Rather than doing a lot of specific work here, I wonder if we need to fix/improve a pre-existing generic mechanism if things are not working as expected without this.

Reply:
Ah, I didn't know about that. Then I believe it's a useless piece of code. I'll remove it.

Reply:
@demiankatz Fixed.